test/test_scrip.py

```diff
@@ -36,6 +36,32 @@ def test_scrip_is_not_ugrid(self):
         with self.assertRaises(KeyError):
             new_ds['grid_corner_lat']
 
+    def test_scrip_writer(self):
```

Comment:
Neither of these two test cases below seems to be testing against actual values, which definitely needs to be the case. I think the current implementation would give us errors at least for `lat_nodes` and `lon_nodes` calculations (i.e. they should be `double` but our calculation converts them to `int`, which is described in one of the comments below)

If we don't have a SCRIP file for that we have a corresponding UGRID file, I think the way for testing should be:

- Read in a SCRIP file and save it as a UGRID file
- Read in that saved UGRID file, and save it as a SCRIP file
- Compare all of the values in that output SCRIP file, which you calculated in `write_scrip`, to the ones in the original input SCRIP file

Reply:
I updated this test, let me know if there should be any further changes to it. I'll mark this as resolved if it looks good to you